api/src/main/java/com/github/retrooper/packetevents/netty/buffer/ByteBufHelper.java

```diff
@@ -239,29 +240,82 @@ public static Object resetWriterIndex(Object buffer) {
         return PacketEvents.getAPI().getNettyManager().getByteBufOperator().resetWriterIndex(buffer);
     }
 
-    public static int readVarInt(Object buffer) {
-        int value = 0;
-        int length = 0;
-        byte currentByte;
-        do {
-            currentByte = readByte(buffer);
-            value |= (currentByte & 0x7F) << (length * 7);
-            length++;
-            if (length > 5) {
-                throw new RuntimeException("VarInt is too large. Must be smaller than 5 bytes.");
-            }
-        } while ((currentByte & 0x80) == 0x80);
-        return value;
-    }
-
-    public static void writeVarInt(Object buffer, int value) {
-        while (true) {
-            if ((value & ~0x7F) == 0) {
-                writeByte(buffer, value);
-                return;
-            }
-            writeByte(buffer, (value & 0x7F) | 0x80);
-            value >>>= 7;
+    public static int getIntLE(Object buffer, int readerIndex) {
+        return PacketEvents.getAPI().getNettyManager().getByteBufOperator().getIntLE(buffer, readerIndex);
+    }
+
+    //Src: https://github.com/jonesdevelopment/sonar/blob/main/common/src/main/java/xyz/jonesdev/sonar/common/fallback/netty/FallbackVarInt21FrameDecoder.java#L69
```

Comment:
this is definetely not the source of this code
either [directly look at velocity](https://github.com/PaperMC/Velocity/blob/9cfcfcf2ed5712e792114a3ab824670e25e23526/proxy/src/main/java/com/velocitypowered/proxy/protocol/netty/MinecraftVarintFrameDecoder.java#L82), or see commit message: https://github.com/jonesdevelopment/sonar/commit/26c31ceeaffdc3c1d3235bd20af18db78c042553
also, please at least mention the license of this piece of code, even if it is GPL

Reply:
Ah sorry, didn't see that